File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.cpp
class Solution {
public:
    static bool mysort(vector<int> &a, vector<int> &b){
        if(a[1]!=b[1]) return a[1]<b[1];
        else return a[0]<b[0];
        
    }
    int findLongestChain(vector<vector<int>>& pairs) {
        sort(pairs.begin(), pairs.end(), mysort);
        int cnt=0, prev=INT_MIN;
        for(auto i: pairs){
            if(i[0]>prev){
                prev=i[1];
                cnt++;
            }
        }
        return cnt;
    }
};

File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.java
class Solution {
public:
    int findLongestChain(vector<vector<int>>& pairs) {
        sort(pairs.begin(), pairs.end(), [&](const vector<int> &a, const vector<int> &b){
            return a[1]<b[1];
        });
        int prev=pairs[0][1], cnt=1;
        for(int i=1; i<pairs.size(); i++){
            if(pairs[i][0]> prev){
                cnt++;
                prev= pairs[i][1];
            }
        }
        return cnt;
    }
};